contracts/CTKNCrowdsale.sol

```diff
@@ -44,4 +75,71 @@ contract CTKNCrowdsale is CappedCrowdsale, RefundableCrowdsale, MintedCrowdsale
         dollarRate = _dollarRate;
         DollarRateSet(dollarRate);
     }
+
+    /**
+     *  Get the amount the given address has overpaid.
+     *  @param addr The address to check for an overpayment balance
+     *  @return the number of `wei` the address overpaid when buying tokens.
+     */
+    function overpaymentBalance(address addr) external view returns (uint256) {
+        return overpaymentVault.deposited(addr);
+    }
+
+    /**
+     * Investors can claim refunds. This overrides `RefundableCrowdsale` `claimRefund`
+     * 1. if crowdsale is unsuccessful refunds the amount they spent on tokens
+     * 2. it will also refund the additional amount they deposited over what was spent on tokens.
+     */
+    function claimRefund() public {
+        require(isFinalized);
+        if (!goalReached() && vault.deposited(msg.sender) != 0) {
+            vault.refund(msg.sender);
+        }
+        if (overpaymentVault.deposited(msg.sender) != 0) {
+            overpaymentVault.refund(msg.sender);
+        }
+    }
+
+    /**
+     *  Overrides `RefundableCrowdsale` finalization task,
+     *  called when owner calls `finalize()`
+     *  Simply enables refunds on the overpayment vault then invokes `super.finalization()`
+     */
+    function finalization() internal {
+        overpaymentVault.enableRefunds();
+        super.finalization();
+    }
+
+    /**
+     * Overrides `RefundableCrowdsale` fund forwarding.
+     * sends the correct funds to both the vault, and the overpaymentVault.
+     */
+    function _forwardFunds() internal {
+        uint256 depositValue = _getTokenAmount(msg.value).mul(rate);
+        uint256 overpaymentValue = _getOverpaymentAmount(msg.value);
+        vault.deposit.value(depositValue)(msg.sender);
+        overpaymentVault.deposit.value(overpaymentValue)(msg.sender);
+    }
+
+    /**
+     *  In this contract the `rate` represents the number of wei needed to buy one token.
+     *  Therefore this function returns the floor of `_weiAmount` / `rate`.
+     *  Fractions of tokens can not be sold.
+     *  @param _weiAmount Value in wei to be converted into tokens
+     *  @return the whole number of tokens that can be purchased with the specified _weiAmount
+     */
+    function _getTokenAmount(uint256 _weiAmount) internal view returns (uint256) {
+      return uint256(_weiAmount / rate);
+    }
+
+    /**
+     *  We can only issue whole numbers of tokens, so any additional wei.
+     *  needs to be refundable once the crowdsale has closed.
+     *  @param _weiAmount Value in wei to be converted into tokens
+     *  @return the wei amount in excess of what was needed to buy a whole number of tokens.
+     */
+    function _getOverpaymentAmount(uint256 _weiAmount) internal view returns (uint256) {
+      return _weiAmount % rate;
```

Comment:
Is there a simple way to extend safemath library for modulus ?

Reply:
There is no need.